src/main/plugin/iso19139.ca.HNAP/schematron/schematron-rules-common.sch

```diff
@@ -57,6 +57,33 @@
     </xsl:for-each>
   </xsl:function>
 
+  <xsl:key name="groupKey"
+    match="*//gmd:transferOptions/gmd:MD_DigitalTransferOptions/gmd:onLine/gmd:CI_OnlineResource[gmd:linkage/gmd:URL]"
+    use="concat(gmd:linkage/gmd:URL/text(), '|', gmd:description/gco:CharacterString/text())"/>
+
+  <xsl:function xmlns:sch="http://purl.oclc.org/dsdl/schematron"
+    name="geonet:getDuplicateResources"
+    as="xs:string">
+    <xsl:param name="elements" as="element()*"/>
+
+    <xsl:variable name="duplicateResources">
+      <xsl:for-each-group select="$elements"
+        group-by="concat(gmd:linkage/gmd:URL/text(), '|', gmd:description/gco:CharacterString/text())">
+
+        <xsl:if test="count(current-group()) &gt; 1">
+          <xsl:if test="exists(tokenize(current-group()[1]/gmd:description/gco:CharacterString/text(), ';')[3])">
```

Comment:
The results could return 1 are more records.  If there is only one then we don't need a comma but if there are multiple then there should be a comma to separate the items.

Before this if, you may want to add the following.
```
          <xsl:if test="position() &gt; 1">
            <xsl:text>, </xsl:text>
          </xsl:if>
```

Reply:
ok, that works. I can add that
![image](https://github.com/metadata101/iso19139.ca.HNAP/assets/74916635/0db9dd68-6461-4f9f-bb56-89b76039fc4d)